src/hotspot/share/runtime/javaCalls.cpp

```diff
@@ -398,20 +390,35 @@ void JavaCalls::call_helper(JavaValue* result, const methodHandle& method, JavaC
       // the call to call_stub, the optimizer produces wrong code.
       intptr_t* result_val_address = (intptr_t*)(result->get_value_addr());
       intptr_t* parameter_address = args->parameters();
+
+      address entry_point;
+      {
+        // The enter_interp_only_mode use handshake to set interp_only mode
+        // so no safepoint should be allowed between is_interp_only_mode() and call
+        NoSafepointVerifier nsv;
+        if (JvmtiExport::can_post_interpreter_events() && thread->is_interp_only_mode()) {
+          entry_point = method->interpreter_entry();
+        } else {
+          // Since the call stub sets up like the interpreter we call the from_interpreted_entry
+          // so we can go compiled via a i2c.
```

Comment:
Do you think removed comment "Otherwise initial entry method will always run interpreted.", or it's understood and redundant?

Reply:
Yes.  I believe that expression in the 'if'  clearly explain the reason of using interpreter. So comment is redundant.